core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -93,6 +91,49 @@ object Battle {
         }
     }
 
+    private fun tryEarnFromKilling(civUnit:ICombatant, defeatedUnit:MapUnitCombatant){
+        val bonusUniquePlaceholderText = "Earn []% of [] opponent's [] as [] for kills"
+        var bonusUniques = ArrayList<Unique>()
+
+        bonusUniques.addAll(civUnit.getCivInfo().getMatchingUniques(bonusUniquePlaceholderText))
+
+        if (civUnit is MapUnitCombatant) {
+            bonusUniques.addAll(civUnit.unit.getMatchingUniques(bonusUniquePlaceholderText))
+        }
+
+        // As of 3.11.4 This is to be deprecated and converted to "Earn [100]% of [Barbarians] opponent's [Strength] as [Culture] for kills" - keeping it here so that mods with this can still work for now
+        bonusUniques.add(Unique("Earn ["
```

Comment:
This is not a good way of doing it, since you're creating several new Uniques every time the function is run, and each Unique creation contains regex parsing.
If you want to do this with uniques, you can create the uniques as private vals, and add them to the bonusUniques in an if, like I dunno `for(unique in civUnit.getCivInfo().getMatchingUniques("Gain Culture when you kill a barbarian unit")) bonusUniques.add(newBarbarianUnique)

Reply:
Thank you for your advice.

-Compatibility for depecrated unit kill bonus uniques done more efficiently and counted only once (since prior to this PR, it was also counted only once)